store/db.go

```diff
@@ -49,15 +51,17 @@ func newDBLock(dirPath string) (*dbLock, error) {
 }
 
 func (dl *dbLock) lock() error {
+	dl.Lock()
 	if err := dl.fl.ExclusiveLock(); err != nil {
+		dl.Unlock()
```

Comment:
So, I went back to the history, found the reason that we do flock() before mutex is because of #1892 (comment)

But I think this solves the problem.